test/api/route/messages_test.dart

```diff
@@ -436,4 +453,195 @@ void main() {
       });
     });
   });
+
+  group('updateMessageFlags', () {
+    Future<UpdateMessageFlagsResult> checkUpdateMessageFlags(
+      FakeApiConnection connection, {
+      required List<int> messages,
+      required UpdateMessageFlagsOp op,
+      required MessageFlag flag,
+      required Map<String, String> expected,
+    }) async {
+      connection.prepare(json:
+        UpdateMessageFlagsResult(messages: messages).toJson());
+      final result = await updateMessageFlags(connection,
+        messages: messages, op: op, flag: flag);
```

Comment:
This ties together the list of message IDs returned (the modified messages) with the list sent (the target messages). The docs (https://zulip.com/api/update-message-flags) aren't very clear on how those relate to each other, but they don't guarantee that they're the same — in principle only some of the target messages might be modified, and then the returned list should contain only those.

Probably simplest is to let the caller do the `prepare`, like we do in the `getMessages` tests above. And then our test data should make the lists different, to illustrate that they might be.